### PR TITLE
virttest/utils_misc: improve archive_as_tarball

### PR DESCRIPTION
**Pull request by PandaWei**
1) Fix typo (s/amen/name/g)
2) Check if the specified compression is supported
3) Use tarfile.open() instead of tarfile.TarFile()
   Please refer to
   https://docs.python.org/3/library/tarfile.html

### CODE REVIEW

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -820,7 +820,7 @@ def archive_as_tarball(source_dir, dest_dir, tarball_name=None,
 
     If the name of the archive is omitted, it will be taken from the
     source_dir. If it is an absolute path, dest_dir will be ignored. But,
-    if both the destination directory and tarball anem is given, and the
+    if both the destination directory and tarball name are given, and the
```
Comment (clebergnu) [inline virttest/utils_misc.py]: OK,

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -840,6 +840,12 @@ def archive_as_tarball(source_dir, dest_dir, tarball_name=None,
     >>> utils_misc.archive_as_tarball('/tmp', '/net/server/backup',
                                       'host1-tmp', 'gz')
     '''
+    # Refer to https://docs.python.org/3/library/tarfile.html#tarfile.open
+    supported_compression = ['gz', 'bz2', 'xz']
+    if compression not in supported_compression:
+        raise exceptions.TestError("compression %s is not supported method %s"
+                                   % (compression, supported_compression))
+
```
Comment (clebergnu) [inline virttest/utils_misc.py]: xz (lzma) is not really supported on Python 2.6 and 2.7:

```
tarfile.open('tarball.tar.xz', 'r:xz')
---------------------------------------------------------------------------
CompressionError                          Traceback (most recent call last)
<ipython-input-2-4bd68a104a27> in <module>()
----> 1 f = tarfile.open('tarball.tar.xz', 'r:xz')

/usr/lib64/python2.7/tarfile.pyc in open(cls, name, mode, fileobj, bufsize, **kwargs)
   1682                 func = getattr(cls, cls.OPEN_METH[comptype])
   1683             else:
-> 1684                 raise CompressionError("unknown compression type %r" % comptype)
   1685             return func(name, filemode, fileobj, **kwargs)
   1686 

CompressionError: unknown compression type 'xz'
```

Please refer to: https://docs.python.org/2.6/library/tarfile.html#tarfile.open

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -853,8 +859,7 @@ def archive_as_tarball(source_dir, dest_dir, tarball_name=None,
                                               tarball_path))
 
     os.chdir(os.path.dirname(source_dir))
-    tarball = tarfile.TarFile(name=tarball_path, mode='w')
-    tarball = tarball.open(name=tarball_path, mode='w:%s' % compression)
+    tarball = tarfile.open(name=tarball_path, mode='w:%s' % compression)
```
Comment (clebergnu) [inline virttest/utils_misc.py]: OK.

File: virttest/utils_misc.py (hunk 2 above)
Comment (PandaWei) [inline virttest/utils_misc.py]: @clebergnu 
Thanks for your reminder.
I have updated it and Please recheck it.

File: virttest/utils_misc.py [hunk 4]
```diff
@@ -840,6 +840,14 @@ def archive_as_tarball(source_dir, dest_dir, tarball_name=None,
     >>> utils_misc.archive_as_tarball('/tmp', '/net/server/backup',
                                       'host1-tmp', 'gz')
     '''
+    mode_str = 'w:%s' % compression
+    help_doc = process.system_output("python -c 'import tarfile;"
+                                     "help(tarfile.open)' | tee",
+                                     ignore_status=True)
```
Comment (clebergnu) [inline virttest/utils_misc.py]: You are probably better just using:

```
import tarfile
if mode_str not in tarfile.open.__doc__:
...
```

Comment (PandaWei) [inline virttest/utils_misc.py]: Thanks a lot,
you idea looks better than me.

**Comment by PandaWei** [comment]
Update

**Pull request merged**